src/crimson/osd/object_context.h

```diff
@@ -75,6 +75,16 @@ class ObjectContext : public ceph::common::intrusive_lru_base<
 
   ObjectContext(hobject_t hoid) : lock(hoid),
                                   obs(std::move(hoid)) {}
+  ObjectContext(const ObjectContext &other)
+    : obs(other.obs),
+      ssc(new SnapSetContext(*other.ssc))
+  {}
+
+  ObjectContext& operator=(const ObjectContext &other) {
+    obs = other.obs;
+    ssc = new SnapSetContext(*other.ssc);
+    return *this;
+  }
```

Comment:
Since rollbacker is the only user, should we introduce a dedicated `get_obc_rollbacker` instead of overriding operator= and a copy ctor? Allowing this may confuse future users.
```
// We may need to rollback the ObjectContext on failed op execution.
// Copy the current obc before mutating it in order to recover on failures.
ObjectContext get_obc_rollbacker (const ObjectContext &other) {
```